### PR TITLE
GPU accelerated rendering & Mouse-handler optimization

### PR DESCRIPTION
**Pull request by LeandroTreu**
- Fixed issue that prevented GPU acceleration for rendering
- Further optimized mouse-handler hover computation
- Moved summarize edges setting to performance section
- Added Arrow LOD checks for deferred edges
- Replaced ellipse drawing with native call

### CODE REVIEW

File: src/renderer/renderer_elements.ts [hunk 1]
```diff
@@ -3184,19 +3173,23 @@ export function drawOctagon(
...
-    ctx.moveTo(x, ym);
-    ctx.bezierCurveTo(x, ym - oy, xm - ox, y, xm, y);
-    ctx.bezierCurveTo(xm + ox, y, xe, ym - oy, xe, ym);
-    ctx.bezierCurveTo(xe, ym + oy, xm + ox, ye, xm, ye);
-    ctx.bezierCurveTo(xm - ox, ye, x, ym + oy, x, ym);
+
+    // CanvasRenderingContext2D ellipse call as an alternative.
+    ctx.ellipse(x+w/2, y+h/2, w/2, h/2, 0, 0, 2 * Math.PI);
+
+    // const kappa = .5522848,
```
Comment (phschaad) [inline src/renderer/renderer_elements.ts:3180]: probably makes sense to remove the commented out code

**Review by phschaad — CHANGES_REQUESTED** [review]

File: src/renderer/canvas_manager.ts [hunk 2]
```diff
@@ -222,6 +222,9 @@ export class CanvasManager {
         this.indices = [];
     }
 
+    // WARNING: This function uses ctx.getImageData() which forces 
+    // the browser to turn off GPU accelerated canvas painting!
+    // It needs to be reworked to not use getImageData() anymore.
```
[the remark this answers is not in the document]
Comment (LeandroTreu) [reply src/renderer/canvas_manager.ts:227]: Done

File: src/renderer/renderer_elements.ts (hunk 1 above)
Comment (LeandroTreu) [reply src/renderer/renderer_elements.ts]: Done

File: src/utils/sdfv_settings.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export class SDFVSettings {
         'showStateNames': true,
         'showMapSchedules': true,
         'showDataDescriptorSizes': false,
-        'summarizeLargeNumbersOfEdges': false,
+        'summarizeLargeNumbersOfEdges': true,
```
[the remark this answers is not in the document]
Comment (LeandroTreu) [reply src/utils/sdfv_settings.ts]: Makes sense!

**Pull request merged**